sdk/core/Azure.Core/src/AzureExtensions.cs

```diff
@@ -1,17 +1,28 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+using System;
 using System.Threading;
 using System.Threading.Tasks;
 
 namespace Azure
 {
     public static class AzureExtensions
     {
-        public static async ValueTask<Response<T>> WaitCompletionAsync<T>(this Task<Operation<T>> operation, CancellationToken cancellationToken = default) where T : notnull
+        public static async ValueTask<Response<T>> WaitForCompletionAsync<T, TOperation>(this Task<TOperation> operation, CancellationToken cancellationToken = default)
+            where T : notnull
+            where TOperation: Operation<T>
         {
             Operation<T> o = await operation.ConfigureAwait(false);
-            return await o.WaitCompletionAsync(cancellationToken).ConfigureAwait(false);
+            return await o.WaitForCompletionAsync(cancellationToken).ConfigureAwait(false);
+        }
+
+        public static async ValueTask<Response<T>> WaitForCompletionAsync<T, TOperation>(this Task<TOperation> operation, TimeSpan? interval, CancellationToken cancellationToken)
```

Comment:
Why make the interval nullable?  They could just call the other overload if they didn't want to pass it.

Reply:
Great point, fixed.